File: app/src/main/java/net/ivpn/client/ui/startonboot/StartOnBootViewModel.java
package net.ivpn.client.ui.startonboot;

import androidx.databinding.ObservableBoolean;
import android.widget.CompoundButton;

import net.ivpn.client.common.prefs.Settings;

import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

import javax.inject.Inject;

public class StartOnBootViewModel {
    private static final Logger LOGGER = LoggerFactory.getLogger(StartOnBootViewModel.class);

    public final ObservableBoolean startOnBoot = new ObservableBoolean();
    public CompoundButton.OnCheckedChangeListener enableStartOnBoot = (compoundButton, value) -> enableStartOnBoot(value);

    private Settings settings;

    @Inject
    StartOnBootViewModel(Settings settings) {
        this.settings = settings;
    }

    void onResume() {
        startOnBoot.set(isStartOnBootEnabled());
    }

    private void enableStartOnBoot(boolean value) {
        LOGGER.info("Start on boot is enabled: " + value);
        startOnBoot.set(value);
        settings.enableStartOnBoot(value);
    }

    private boolean isStartOnBootEnabled() {
        return settings.isStartOnBootEnabled();
    }
}

File: app/src/main/java/net/ivpn/client/v2/viewmodel/StartOnBootViewModel.kt
package net.ivpn.client.v2.viewmodel

/*
 IVPN Android app
 https://github.com/ivpn/android-app

 Created by Oleksandr Mykhailenko.
 Copyright (c) 2020 Privatus Limited.

 This file is part of the IVPN Android app.

 The IVPN Android app is free software: you can redistribute it and/or
 modify it under the terms of the GNU General Public License as published by the Free
 Software Foundation, either version 3 of the License, or (at your option) any later version.

 The IVPN Android app is distributed in the hope that it will be useful,
 but WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY
 or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for more
 details.

 You should have received a copy of the GNU General Public License
 along with the IVPN Android app. If not, see <https://www.gnu.org/licenses/>.
*/

import androidx.databinding.ObservableBoolean
import net.ivpn.client.common.BuildController
import net.ivpn.client.common.dagger.ApplicationScope
import net.ivpn.client.common.prefs.Settings
import javax.inject.Inject

@ApplicationScope
class StartOnBootViewModel @Inject constructor(
        val settings: Settings,
        buildController: BuildController
) {
    val isStartOnBootEnabled = ObservableBoolean()
    val isStartOnBootSupported = ObservableBoolean()

    init {
        isStartOnBootSupported.set(buildController.isStartOnBootSupported)
    }

    fun onResume() {
        isStartOnBootEnabled.set(settings.isStartOnBootEnabled)
    }

    fun reset() {
        isStartOnBootEnabled.set(settings.isStartOnBootEnabled)
    }
}